lib/device/fan.js

```diff
@@ -125,107 +203,137 @@ export default class {
+          this.cacheSpeed = 'SPEED_5';
+          this.service.updateCharacteristic(this.hapChar.RotationSpeed, 5 * 12.5);
 
           // Log the change if appropriate
           if (this.enableLogging) {
-            this.log('[%s] %s [%s].', this.name, platformLang.curSpeed, 'medium')
+            this.log('[%s] %s [%s].', this.name, platformLang.curState, this.cacheSpeed);
           }
         }
-        break
+        break;
       }
+      case fanState.SPEED_6: {
+        // Set the speed to 6
+        if (this.cacheSpeed !== 'SPEED_6') {
+          this.cacheSpeed = 'SPEED_6';
+          this.service.updateCharacteristic(this.hapChar.RotationSpeed, 6 * 12.5);
 
-      case 'MwUDAAAAAAAAAAAAAAAAAAAAADU=': {
-        // High speed
-        // Check if the device was previously OFF, if so update HomeKit to show as ON
-        if (this.cacheState === 'off') {
-          // Update Homekit and the cache
-          this.service.updateCharacteristic(this.hapChar.On, true)
-          this.cacheState = 'on'
+          // Log the change if appropriate
+          if (this.enableLogging) {
+            this.log('[%s] %s [%s].', this.name, platformLang.curState, this.cacheSpeed);
+          }
+        }
+        break;
+      }
+      case fanState.SPEED_7: {
+        // Set the speed to 7
+        if (this.cacheSpeed !== 'SPEED_7') {
+          this.cacheSpeed = 'SPEED_7';
+          this.service.updateCharacteristic(this.hapChar.RotationSpeed, 7 * 12.5);
 
           // Log the change if appropriate
           if (this.enableLogging) {
-            this.log('[%s] %s [%s].', this.name, platformLang.curState, this.cacheState)
+            this.log('[%s] %s [%s].', this.name, platformLang.curState, this.cacheSpeed);
           }
         }
-        if (this.cacheSpeed !== 99) {
-          this.cacheSpeed = 99
-          this.service.updateCharacteristic(this.hapChar.RotationSpeed, this.cacheSpeed)
+        break;
+      }
+      case fanState.SPEED_8: {
+        // Set the speed to 8
+        if (this.cacheSpeed !== 'SPEED_8') {
+          this.cacheSpeed = 'SPEED_8';
+          this.service.updateCharacteristic(this.hapChar.RotationSpeed, 8 * 12.5);
 
           // Log the change if appropriate
           if (this.enableLogging) {
-            this.log('[%s] %s [%s].', this.name, platformLang.curSpeed, 'high')
+            this.log('[%s] %s [%s].', this.name, platformLang.curState, this.cacheSpeed);
           }
         }
-        break
+        break;
       }
-      case 'MxgBAAAAAAAAAAAAAAAAAAAAACo=': {
-        // Start oscillation
-        if (this.cacheDir !== 'on') {
-          this.cacheDir = 'on'
-          this.service.updateCharacteristic(this.hapChar.RotationDirection, 1)
+      case fanState.SWING_ON: {
+        // Set the swing to on
+        if (this.cacheSwing !== 'SWING_ON') {
+          this.cacheSwing = 'SWING_ON';
+          this.service.updateCharacteristic(this.hapChar.SwingMode, true);
```

Comment:
SwingMode uses integer value not boolean value:
https://developers.homebridge.io/#/characteristic/SwingMode

So needs to be updated with `1` rather than `true`, and same for line 327 with `0`